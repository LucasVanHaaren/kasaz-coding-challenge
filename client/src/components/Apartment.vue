<template>
  <v-card class="mx-auto" max-width="700" outlined>
    <v-img :aspect-ratio="16/9" class="white--text align-end" height="300px" :src="thumbnail">
      <div class="ml-5 mb-2">
        <h1>{{ price }}€</h1>
        <h3>{{ pricePerSqm }}€/m²</h3>
      </div>
    </v-img>
    <v-card-title>{{ title }}</v-card-title>
    <v-divider></v-divider>
    <v-card-actions>
      <div>
        <v-icon>fas fa-check-circle</v-icon>
        {{ sqm }} m²
      </div>
      <v-spacer></v-spacer>
      <v-divider class="mx-4" inset vertical></v-divider>
      <v-spacer></v-spacer>
      <div>
        <v-icon>fas fa-bed</v-icon>
        {{ bedroomsCount }} habs.
      </div>
      <v-spacer></v-spacer>
      <v-divider class="mx-4" inset vertical></v-divider>
      <v-spacer></v-spacer>
      <div>
        <v-icon>fas fa-bath</v-icon>
        {{ bathroomsCount }} baños
      </div>
    </v-card-actions>
  </v-card>
</template>

<script>

export default {
  name: 'Apartment',
  computed: {
      pricePerSqm: function () {
          return Math.floor(this.price / this.sqm);
      }
  },
  props: {
      title: {
          type: String,
          required: true
      },
      price: {
          type: Number,
          required: true
      },
      sqm: {
          type: Number,
          required: true
      },
      bedroomsCount: {
          type: Number,
          required: true
      },
      bathroomsCount: {
          type: Number,
          required: true
      },
      thumbnail: {
          type: String,
          required: false,
          default: "/pictures/default.jpg"
      }
  }
};
</script>