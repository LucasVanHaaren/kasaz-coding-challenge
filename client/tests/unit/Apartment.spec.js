import { shallowMount } from '@vue/test-utils'
import Apartment from '@/components/Apartment.vue'

describe('Apartment.vue', () => {

  const props = {
    title: "Title",
    price: 150000,
    sqm: 100,
    bedroomsCount: 4,
    bathroomsCount: 2
  };

  it('match snapshot', () => {
    const wrapper = shallowMount(Apartment, {
      propsData: props
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const wrapper = shallowMount(Apartment, {
      propsData: props
    });
    const res = props.price / props.sqm;
    expect(wrapper.find('h3').text()).toBe(`${res}€/m²`);
  });
})
