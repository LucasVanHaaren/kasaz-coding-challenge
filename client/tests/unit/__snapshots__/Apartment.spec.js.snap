// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apartment.vue match snapshot 1`] = `
<v-card
  class="mx-auto"
  max-width="700"
  outlined=""
>
  <v-img
    aspect-ratio="1.7777777777777777"
    class="white--text align-end"
    height="300px"
    src="/pictures/default.jpg"
  >
    <div
      class="ml-5 mb-2"
    >
      <h1>
        150000€
      </h1>
       
      <h3>
        1500€/m²
      </h3>
    </div>
  </v-img>
   
  <v-card-title>
    Title
  </v-card-title>
   
  <v-divider />
   
  <v-card-actions>
    <div>
      <v-icon>
        fas fa-check-circle
      </v-icon>
      
      100 m²
    
    </div>
     
    <v-spacer />
     
    <v-divider
      class="mx-4"
      inset=""
      vertical=""
    />
     
    <v-spacer />
     
    <div>
      <v-icon>
        fas fa-bed
      </v-icon>
      
      4 habs.
    
    </div>
     
    <v-spacer />
     
    <v-divider
      class="mx-4"
      inset=""
      vertical=""
    />
     
    <v-spacer />
     
    <div>
      <v-icon>
        fas fa-bath
      </v-icon>
      
      2 baños
    
    </div>
  </v-card-actions>
</v-card>
`;
